src/modeler/AutoRest.Swagger.Tests/SwaggerModelerValidationTests.cs

```diff
@@ -40,6 +40,16 @@ private IEnumerable<ValidationMessage> ValidateSwagger(string input)
             return messages;
         }
 
+        /// <summary>
+        /// Verifies that a clean Swagger file does not result in any validation errors
+        /// </summary>
+        [Fact]
+        public void CleanFileValidation()
+        {
+            var messages = ValidateSwagger(Path.Combine("Swagger", "Validation", "clean-complex-spec.json"));
+            Assert.Equal(0, messages.Where(m => m.Severity >= LogEntrySeverity.Warning).Count());
```

Comment:
You might consider :

``` csharp
Assert.Empty(messages.Where(m => m.Severity >= LogEntrySeverity.Warning)); 
```

Which has the virtue, that if the collection takes a lot of time to count all the elements, assert will fail faster.